### ISSUE TITLE
pgbouncer=true flag not working as expected

### ISSUE THREAD

**Issue by aarshaw** [prisma/prisma#2907]
## Bug description
**Related thread**
https://github.com/prisma/prisma-client-js/issues/651#issuecomment-652439460

**Using DigitalOcean Connection Pools**
<img width="602" alt="Screenshot 2020-07-01 at 15 09 17" src="https://user-images.githubusercontent.com/5471723/86254626-fefdbe80-bbad-11ea-8612-d73478730c35.png">

**Connection String**
```
postgresql://xxxxxxx:xxxxxxx@xxxxxxx:25061/Pool?sslmode=require&pgbouncer=true&schema=xxxxx
```

## How to reproduce

- Create new DO managed postgres db 
- Deploy Prisma schema (without pgbouncer)
- Add connection pool
- Connect using the connection pool string above (with pgbouncer)
- Run app/prisma studio - execute 3 or 4 actions

Results in this error:
 ```
Error: Error occurred during query execution:
ConnectorError(ConnectorError { user_facing_error: None, kind: QueryError(Error { kind: Db, cause: Some(DbError { severity: "ERROR", parsed_severity: Some(Error), code: SqlState("26000"), message: "prepared statement \"s162\" does not exist", detail: None, hint: None, position: None, where_: None, schema: None, table: None, column: None, datatype: None, constraint: None, file: Some("prepare.c"), line: Some(512), routine: Some("FetchPreparedStatement") }) }) })
    at PrismaClientFetcher.request (/Users/aaron/.cache/prisma/studio/Users-aaron-Desktop-Ember/runtime/index.js:1:185866)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
```

## Prisma information

Fairly large schema (can't share specifics)
Using nexus/schema (although reproducible in prisma studio also)

## Environment & setup

- OS: Mac OS
- Database: PostgreSQL
- Prisma version: 2.1.3
- Node.js version: 12.16.3

**Comment by janpio** [comment]
Thanks for the information here. 

We have e2e tests for DO's PgBouncer, so we are sure it works fundamentally: https://github.com/prisma/e2e-tests/tree/dev/databases/digitalocean-pgbouncer 

The first and only difference that pops up for me is that our database connection is not using `sslmode=require` and does not specify a schema via `&schema` - but that should not make a difference.

We will look into this.

**Comment by aarshaw** [comment]
I can confirm I'm experiencing the same behaviour without the `sslmode` and `schema` query params - also reverted to `2.0.0` and had the same issue

**Comment by divyenduz** [comment]
@aarshaw Did you re-generate the client after adding `pgbouncer=true`, I am unable to reproduce this issue.

**Comment by aarshaw** [comment]
Yeah, deleted node_modules, regenerated client etc. - the connection string is provided by an env var is that an issue?

**Comment by aarshaw** [comment]
What I'm seeing in the tests is that you're executing one query with and without the param - this works for me. Where it isn't working is after a few seconds usage - presumably, after it executes more than one query - have attached a quick screen recording to illustrate -

[recording.mov.zip](https://github.com/prisma/prisma/files/4863047/recording.mov.zip)

**Comment by divyenduz** [comment]
Thanks, in my local reproduction, I did try multiple queries (even some load testing). Thanks for the video, I will try to give it another go.

**Comment by aarshaw** [comment]
Thanks - let me know if you need any further info relating to the environment/schema to replicate

**Comment by divyenduz** [comment]
@aarshaw Are you able to reproduce this without using studio? By hitting a script/endpoint multiple times?

**Comment by aarshaw** [comment]
Yeah - I'm able to reproduce using:
```
import { PrismaClient } from "@prisma/client";

const prisma = new PrismaClient();

const run = async () => {
  await prisma.xxxxx.findMany({
    take: 100,
    select: {
      ....
    }
  });

  await prisma.yyyyy.findMany({
    take: 100,
    select: {
      ....
    }
  });

  await prisma.zzzzz.findMany({
    take: 100,
    select: {
      ....
    }
  });

  await prisma.aaaaa.findMany({
    take: 100,
    select: {
      ....
    }
  });
};

run();
```

Where it crashes on the third execution (zzzzz)

**Comment by divyenduz** [comment]
Thank you, is it possible for you to share your schema and some seed data at `schemas@prisma.io`. We are happy to sign an NDA if you want. With that, a minimal reproduction would be much easier for us.

**Comment by divyenduz** [comment]
Thank you @aarshaw for sharing the schema based on the information we were able to create an isolated minimal reproduction: https://github.com/divyendu-test/prisma-2907

This has something to do with a relation and enums on both sides of the relations used in conjunction with pgbouncer.

**Comment by pantharshit00** [comment]
I can also reproduce this

**Comment by divyenduz** [comment]
Update: The underlying issue is fixed and will be released with `2.4.0` (next week). If you want to try it out now, you can use `npm install -g @prisma/cli@dev`
